### DISCUSSION TITLE
✅ Add Twitter "X" icon to the Icon library

### DISCUSSION THREAD

**Discussion by cswebstuff** [elementor/elementor#23345] [Feature Request]
UPDATE TO MY POST:

**TWITTER X CUSTOM ICON FOR ELEMENTOR**

Hey Elementor friends,

I made a download page to share the Twitter X icon I made for Elementor.

It's a safe custom created Fontello zip file with the official X logo downloaded from Twitters "Brand Toolkit" page. I'll be posting tons more useful Elementor codes, CSS, tips and solutions on CodeXYZ.com so give it a bookmark to see what I'll be adding soon. 🤓

https://codexyz.com/twitter-x-icon-for-elementor/
__________________________________________________________

### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

Trying to stay up-to-date with the new Twitter icon "X" - it's not available yet in the Elementor Font Awesome library (as of 8/8/23)

### Describe the solution you'd like

Provide the new Twitter X icon for social media

### Describe alternatives you've considered

I have to currently make and upload my own SVG   :-(

**Comment by Clockworkcreative** [comment]
I completely agree! This is really frustrating and whilst I have uploaded my own SVG for the time being, for the social share widget on posts it automatically assigns the icon so I can't use my custom SVG. 

In addition, the Threads logo needs to be added. Both the Twitter 'X' icon and Threads are available on Font Awesome 6.

**Comment by portrayaloflife** [comment]
Font Awesome released it but it's not in Elementors icons yet 
https://fontawesome.com/icons/x-twitter?f=brands&s=solid

**Comment by schmidtdesign** [comment]
Looks like we are still waiting for the x-twitter icon to be available in elementor's icon library - why?

**Reply by cswebstuff** [reply]
Hey schmidtdesign,

I have a fontello zip to add that Twitter X to your Elementor custom icon library. Let me know if you want it, just message me and I'll hang it off my site for you to download. 

By the way, those icon SVG zip files must be in a proprietary format because I tried zipping a .svg file and Elementor will not accept it.  Only zipped files from the actual companies of either:  Fontello, IcoMoon or Fontastic will work.

**Reply by TehKonnos** [reply]
When I tried to upload the X SVG icon in the library, it looked a bit strange next to the other social icons. If your file solves this issue, I am also interested in downloading it.

**Reply by onlinepacific** [reply]
There's a temporary and easy solution.

1. Download the X SVG file with the color you want from FontAwesome, Flaticon or whatever you want.
2. Add it to the medias of your website.
3. Add this CSS with the link of your file, and ajust width if necessary
i.fab.fa-twitter {
content: url(wp-content/uploads/twitter-alt-2.svg);
}
4. Save the code.

This will replace the Twitter icon by the X icon downloaded in your library

**Reply by prcptnmrktng** [reply]
@onlineorg Works like a charm, thanks so much!

**Reply by cswebstuff** [reply]
Hi @pkimmel6345,

Did you use my SVG download page(s)?  If yes which one:  The Twitter X (or) the Meta Threads?  Just curious.  

Thx,
@cswebstuff

**Comment by hitsgee** [comment]
Elementor are still running Font Awesome 5 despite version 6 being out for nearly 2 years. We are missing many new and improved icons while they take their sweet time to upgrade.

Elementor added FA5 in Elementor 2.6 we are on the beta for 3.17 and still no FA6.

**Comment by Clockworkcreative** [comment]
It's pretty poor show! Although you can upload the SVGs in the meantime for generic social icons, for the social share button widget where you can't easily upload an alternative icon. You only have a pre-defined list and it automatically applies the platform icon. Pull your finger out Elementor!

**Comment by cswebstuff** [comment]
**TWITTER X CUSTOM ICON FOR ELEMENTOR**

Hey Elementor friends,

I made a download page to share the Twitter X icon I made for Elementor.  

It's a safe custom created Fontello zip file with the official X logo downloaded from Twitters "Brand Toolkit" page.  I'll be posting tons more useful Elementor codes, CSS, tips and solutions on CodeXYZ.com so give it a bookmark to see what I'll be adding soon.  🤓

https://codexyz.com/twitter-x-icon-for-elementor/

**Reply by ankejespers** [reply]
Thanks for this!!

**Reply by cswebstuff** [reply]
Glad you found it useful.  If you have time please let people know it's a safe download.  :-)

**Reply by onlinepacific** [reply]
There's a temporary and easy solution.

1. Download the X SVG file with the color you want from FontAwesome, Flaticon or whatever you want.
2. Add it to the medias of your website.
3. Add this CSS with the link of your file, and ajust width if necessary
i.fab.fa-twitter {
content: url(wp-content/uploads/twitter-alt-2.svg);
}
4. Save the code.

This will replace the Twitter icon by the X icon downloaded in your library

**Reply by cswebstuff** [reply]
Or just download from my link, I did all the work already and it's the official Twitter X logo from their website, no CSS needed just install using Elementor's custom icon insert tool.  ;-)

**Reply by LitBearTea** [reply]
Thanks a lot ! It took me one minute to make it works. I'll follow you...

**Reply by cswebstuff** [reply]
Awesome, glad it was helpful and easy to use.  😊

**Reply by thehels** [reply]
Thank you for saving us the time to do this!!

**Reply by cswebstuff** [reply]
Hi @thehels, my pleasure!  I can't wait to share all of the other tips and goodies I have saved up for everyone on my new CodeXYZ.com site.  I just have to find time to put all that info into useful blog articles.  Looking forward to it though, I have over 40+ helpful tips and instructions (the hard to find stuff) saved up for everyone in files!  :-)

**Reply by Ant7t5** [reply]
This fix is awesome thanks cs. I don't suppose you have one for Threads do you? 👍 Ant

**Reply by cswebstuff** [reply]
Hey thanks Ant.

I'm working on getting the official Threads SVG (crazy difficult to find) - it's a work in progress but I have communication out to the Meta branding department and will make an icon as soon as I get the "official" file. 

I'll post it here when it's ready.

**Reply by cswebstuff** [reply]
**META THREADS CUSTOM ICON FOR ELEMENTOR**

Hi All,

OK, it's here...  I created a Meta Threads Icon for Elementor for you to download.  Installation for Elementor custom icon is included on the page.  

This is a safe custom created Fontello zip download file for the official Meta Threads logo. The logo is from Meta's Brand Resources page. I'll be posting tons more useful Elementor codes, CSS, tips and solutions on CodeXYZ.com so don't forget to bookmark CodeXYZ.com to see what's added next. 🤓

https://codexyz.com/meta-threads-icon-for-elementor 

Enjoy!

**Reply by ankur3232** [reply]
After uploading, it is showing correct thumbnail in media, but while choosing it from social icons in elementor on my page, the icon becomes a "home" icon. Its not working for me. Can yo help me out here please

**Reply by cswebstuff** [reply]
@ankur3232
Just so we don't spam this forum with help replies, signup to the newsletter on CodeXYZ.com so I can email you directly and I'll see if I can help.

**Comment by onlinepacific** [comment]
There's a temporary and easy solution.

1. Download the X SVG file with the color you want from FontAwesome, Flaticon or whatever you want.
2. Add it to the medias of your website.
3. Add this CSS with the link of your file, and ajust width if necessary
i.fab.fa-twitter {
content: url(wp-content/uploads/twitter-alt-2.svg);
}
4. Save the code.

This will replace the Twitter icon by the X icon downloaded in your library

**Comment by hiyakake** [comment]
I understand that they are available by uploading an SVG, but Elementor should be easy for everyone.
We would very much like to be able to use the new icons as soon as possible without having to use complicated methods.
In fact, even if you upload an SVG, it will not display beautifully unless the margins of the SVG are the same size as those of "FontAwesome".
Adjusting SVG margins may seem simple, but it is not a skill that everyone has.

**Reply by Clockworkcreative** [reply]
Exactly! Elementor has an icon widget and also uses social icons within other widgets, you would expect them to be relatively up-to-date and not require workarounds. In July, X replaced Twitter and Threads was launched and they've been available in FontAwesome for months! This should absolutely have been part of a standard update to an existing functionality, we shouldn't have to ask for this as a feature request!

**Comment by websubst** [comment]
The "share buttons" widget doesn't let us choose the icon X logo for Twitter

**Reply by cswebstuff** [reply]
interesting, let me take a look.  Elementor should still include our custom icons in that widget, if not I'll add that to their "to-do" list.  ;-)

**Reply by websubst** [reply]
Thank you in advance👍

**Reply by cswebstuff** [reply]
@blackvx, 
I'm gonna try to write some custom code for that - it's an Elementor specific list and unfortunately has no ties to the icon library.  Something they should definitely add for customizations.  

I have a temporary hack for you - not fantastic (and only if you're using the "icon & text") for the share buttons, but if you click on your Twitter tab in the "Edit Share Buttons" widget - in the "Custom Label" field type in a capital "X" (without the quotes).  This will give you the bird icon and an X which may be helpful to some people showing both.  😏

 But for a quick temp fix this might be helpful.
 
 Apparently there is already a Github request to use custom icons in the social share button widget - since July 4, 2022!
 https://github.com/elementor/elementor/issues/19078  so it doesn't look like that's happening from their end very soon.

**Reply by websubst** [reply]
Yikes! Thanks for trying! I'll replace the share buttons widget with something else instead. I prefer the bird logo to a plain X :-)

**Reply by cswebstuff** [reply]
Hey, if you don't mind circling back let us know what you find - it would be helpful for everyone here.  :-)

**Reply by ellieender** [reply]
@cswebstuff @blackvx Any updates? I'm also experiencing this on my end with my blog site. A colleague sent this over: https://support.buddyboss.com/support/solutions/articles/72000610217-how-to-change-the-twitter-social-share-icon-to-the-new-x-logo

But I don't think this code works for this situation.

**Reply by cswebstuff** [reply]
Hey @ellieender,
Thanks for the reminder and for posting that link, it was a helpful place to start but you're right it won't do the trick for Elementor's social share buttons.  I was able to code 2 working versions today with Square & Round icons, and should have the page & code up by end of day tomorrow (January 11).  I'll update this message when it's ready.

**Reply by cswebstuff** [reply]
OK, my Twitter X Social Share Button for Elementor is finally done!  
https://codexyz.com/twitter-x-social-share-button-for-elementor/

There are 2 options available...  The EASY way and the more DIFFICULT stylish option.  
*** Please pound that LIKE button at the bottom of that page if this helps you out! ***

Hit me up here if you have any questions or helpful code tweaks to this!  🤓

**Comment by m1ga** [comment]
https://github.com/elementor/elementor/pull/24315
it contains `x-twitter` :+1:

**Reply by Clockworkcreative** [reply]
It's good to see Elementor are finally doing something about this! Although I can't see any timeline for deployment, hopefully it's soon!

**Comment by toineenzo** [comment]
FYI: FA6 is now added to the roadmap
<img width="302" alt="image" src="https://github.com/elementor/elementor/assets/4921397/81143963-bf96-4367-b5a3-8963321f5816">

**Reply by m1ga** [reply]
They are working on it since November. You can follow the PR https://github.com/elementor/elementor/pull/24315

**Reply by toineenzo** [reply]
I know :)

**Comment by cswebstuff** [comment]
In the meantime people can download the custom created icon from https://codexyz.com and follow the instructions there.  😎

**Comment by CenterStepMarketing** [comment]
@cswebstuff I tried downloading your zip file from codexyz but am getting this error. File was not altered or renamed in any way; just straight download from your original link and uploaded to my site(s). Elementor and Elementor Pro are both on their latest release (3.19.0).

Strange part is that it did work on the first website I attempted, but after that, it has not worked on any other sites.

![image](https://github.com/elementor/elementor/assets/110869361/aaccc957-1619-4f1a-bcbe-a00a92b54f51)

**Reply by CenterStepMarketing** [reply]
Hi @cswebstuff,

I can confirm the theme is the same, Elementor Pro is installed, active and linked to a paid account, and the zip file was uploaded without being opened or extracted. I also deleted the original zip file and downloaded a fresh version from your link.

I cleared both object and server cache, and used the two additional "tools" of Regenerate Files & Data" and "Sync Library."

I can also confirm I am **not** uploading the zip file to the Media Library -- I am uploading the zip file to the "Custom Icons" link under Elementor.

Unfortunately, I am still getting this error message. Here is a Loom link to a screenshare walking through both the website where it is working and the website where it is not working to help confirm all the steps above that you mentioned and to confirm that I am taking the correct steps on my end.

https://www.loom.com/share/6781cb781c6842a68f1aec13f4d423e6?src=composer

**Reply by portrayaloflife** [reply]
@cswebstuff Can we stop using this thread to peddle third party stuff, it's awesome you want to help. But this forum isn't to trouble shoot your own plugins. If you want to offer service/solutions please do it outside of this github

**Reply by cswebstuff** [reply]
@portrayaloflife, 

- It's not a plugin just a free file I made available to people who need it. 
- Someone was having a problem so I was being kind and helping him out, I emailed him directly now so no one will be disturbed. 
- I'm not peddling anything. 

I created this original post, feel free to unsubscribe at any time and I hope you found the file helpful for your own projects.

**Reply by portrayaloflife** [reply]
@cswebstuff You are, you're peddling your file to everyone and troubleshooting it. Stop please. People can upload an X SVG easily if they need to. Again, great you want to help just direct them to somewhere off this thread or offer a support email in the file you're sending.

**Reply by cswebstuff** [reply]
Please unsubscribe

**Reply by portrayaloflife** [reply]
Will report you if you keep peddling https://codexyz.com in this github.

**Reply by ellieender** [reply]
Since Elementor has refused, up till today, to provide a solution, what's wrong with developers coming up with their own solutions and sharing it? haha, sheesh.

**Comment by galantes** [comment]
I've solved this by putting a few lines of jQuerry and CSS code in the FOOTER of the page. This is replacing existing Twitter icon and blue color in Elementor with new X icon and black color. 

`<script>
jQuery(document).ready(function($) {
    var newElement = $('<svg xmlns="http://www.w3.org/2000/svg" width="157.66mm" height="147.57mm" viewBox="0 0 5276.1 4938.2"><path d="m12.84 0 2037 2723.7-2049.9 2214.5h461.38l1794.7-1938.9 1450 1938.9h1570l-2151.7-2876.8 1908-2061.3h-461.39l-1652.8 1785.6-1335.4-1785.6h-1570zm678.48 339.82h721.25l3185 4258.5h-721.25l-3185-4258.5z"/></svg>');
    $('.e-fab-twitter').replaceWith(newElement);});
</script>`

`<style>.elementor-share-buttons--color-official.elementor-share-buttons--skin-boxed .elementor-share-btn_twitter .elementor-share-btn__icon, .elementor-share-buttons--color-official.elementor-share-buttons--skin-minimal .elementor-share-btn_twitter .elementor-share-btn__icon {background-color: #000!important;}</style>`

I get this as a result:

![x_elementor](https://github.com/elementor/elementor/assets/29680056/221656df-a272-4009-b730-46d09415b885)

Hope it helps to someone.

**Reply by hitsgee** [reply]
They added the new Twitter and Threads icons to the beta should be out publicly in a week or so.

**Reply by toineenzo** [reply]
I wouldn't do this, it's bad practise

**Comment by leecollings-sb** [comment]
Confused here, as apparently this was added to Elementor on the 11th MArch - however I still cannot use the FA6 X Twitter logo on any page where Elementor icons are used (as it's using the old library still). So what's actually been updated or added?

UPDATE: Actually I can see what they've done. They've just added the X icon to their existing v5.15, rather than that include the new FA6. So I STILL am unable to upgrade my FA theme version to 6, or use the X icon, because the problem still exists where if I try to load the X icon on a page that includes Elementor icons, my theme X icon doesn't display because Elementor is forcing its own FA5 version to load over my theme's FA6.

Why can't Elementor just update the entire FA library it uses to v6?

**Reply by m1ga** [reply]
> Why can't Elementor just update the entire FA library it uses to v6?

follow the FA6 PR as mentioned ealier: https://github.com/elementor/elementor/pull/24315

**Comment by mllemata** [comment]
I have version 3.20.1 of Elementor and 3.20.0 of Elementor Pro, but I can't find the X icon in the list.

**Reply by m1ga** [reply]
![Screenshot_20240318_150744](https://github.com/elementor/elementor/assets/4334997/9c7796ce-4744-477b-8e88-d94f48d24e9b)

**Reply by mllemata** [reply]
![image](https://github.com/elementor/elementor/assets/114259428/fdc32d67-b013-4bef-ac3f-7e4ff8f83555)

**Reply by m1ga** [reply]
which widget are you using? I don't see the "Recommended" tab in my selection box. I'm using the normal icon widget

**Reply by mllemata** [reply]
I'm using the Social Icons widget and also normal Icon widget, but I'm searching in all icons, not just Recommended.

**Reply by m1ga** [reply]
ok, both work for me. Just saying that I don't have the "Recommended" tab at all in my vanilla Elementor 2.20.1 version. But I only have the free version so it might be a Pro thing.

**Comment by nicholaszein** [comment]
Hey everyone!

### As you might already know, the 𝕏 logo for Twitter was implemented with the release of Elementor and Elementor Pro version 3.20.0. 🙌 
You can now use the Icon Library to include the 𝕏 logo on any icon location on your website. 

### Social Icons
For the [Social Icons widget](https://elementor.com/help/social-icons-widget/), you can use the Icon Picker and find the 𝕏 icon in the icon library by typing either **_twitter_** or **`x `** (with a space) in the search box. 

**It's important to note that this icon is only available if you have <ins>purged Object Cache</ins>. Otherwise you won't see the icon in the library!**

### Share Buttons (Pro)
For the [Share Buttons widget (Pro)](https://elementor.com/help/share-buttons-widget-pro/), we used a different approach. To maintain Backward Compatibility we added a new Network called **X** to the Network Dropdown List. You need to manually select that option to see the new icon.

### Backward Compatibility
Elementor takes backward compatibility very serious. We have no intention to make changes to existing features that may break your site. For this reason we decided not to replace all icons with the new 𝕏 icon right away. This ensures all existing sites remain intact, and gives you the option to chose if you want to use the new icon or not.

----

Thank you everyone for your feedback and suggestions! 🙏 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
